launcher/container_runner.go

```diff
@@ -153,7 +176,21 @@ func NewRunner(ctx context.Context, cdClient *containerd.Client, token oauth2.To
 		if err != nil {
 			return nil, fmt.Errorf("failed to get principal tokens: %w", err)
 		}
-		return [][]byte{[]byte(idToken)}, nil
+
+		tokens := [][]byte{[]byte(idToken)}
+
+		if len(launchSpec.ImpersonateServiceAccounts) != 0 {
+			// Fetch impersonated ID tokens.
+			for _, sa := range launchSpec.ImpersonateServiceAccounts {
+				idToken, err := fetchImpersonatedToken(ctx, sa, audience)
+				if err != nil {
+					return nil, fmt.Errorf("failed to get impersonated tokens: %w", err)
```

Comment:
Maybe helpful for debug to include the sa email in the error.

Reply:
Joe commented on one of the attestation verifier CLs that the emails could be PII and to avoid printing them out. If it's the same case here, could instead print the index for the email (which is what we do in the attestation verifier).